### DISCUSSION TITLE
0.4.1

### DISCUSSION THREAD

**Discussion by alienatedsec** [alienatedsec/yi-hack-v5#356] [Announcements]
## Breaking changes !!!
* There is a new baseline `0.4.1`, so you need to prepare your SD card for it - follow https://github.com/alienatedsec/yi-hack-v5/discussions/267 and this guidance  #248 
* The camera web page is no longer under port `8080` but back to default port `80` - access your camera web page using its IP only. Those with proxy environments or `Home Assistant` integrations must amend their settings.
* The new baseline is needed to fix OTA upgrade #326 - cameras on earlier versions cannot upgrade from the WebUi as there is an additional baseline check - it refers to `0.4.1` for the latest release - https://github.com/alienatedsec/yi-hack-v5/commit/23a96a5700bb9616fc56b905e772737ed44870c6 - copy the release file onto your SD card and change its name as per the guidance in #248
* Future non-baseline upgrades will not require a FAT32 SD card preparation

## What's Changed (Generated by GitHub, high-level)
* @roleoroleo testing merge for pre-release by @alienatedsec in https://github.com/alienatedsec/yi-hack-v5/pull/289
* Merging tested commits by @alienatedsec in https://github.com/alienatedsec/yi-hack-v5/pull/299
* Merging onto pre-release by @alienatedsec in https://github.com/alienatedsec/yi-hack-v5/pull/330
* [~] Onvif fixes by @alienatedsec in https://github.com/alienatedsec/yi-hack-v5/pull/331
* fix: WARNING in HA => MQTT entity name starts with the device name in… by @Minims in https://github.com/alienatedsec/yi-hack-v5/pull/327
* fix: mqtt disconnecting immediately after boot by @fidesachates in https://github.com/alienatedsec/yi-hack-v5/pull/328
* Pre release 0.4.1x by @alienatedsec in https://github.com/alienatedsec/yi-hack-v5/pull/341
* Improve MQTT by @Minims in https://github.com/alienatedsec/yi-hack-v5/pull/349
* Pre release 0.4.1x merge to master by @alienatedsec in https://github.com/alienatedsec/yi-hack-v5/pull/355

### Sorry for the below mess - I will try to be consistent with the next releases
* [Restore feature to edit config files](https://github.com/alienatedsec/yi-hack-v5/pull/289/commits/9a8871666fe542898cce1cc3c0dd5d5e5dd36e6f)
* [Add a flag to decrease binary size](https://github.com/alienatedsec/yi-hack-v5/pull/289/commits/5c0e1f6d39f53edc84593ff1727380437532e75b)
* [Remove old patch](https://github.com/alienatedsec/yi-hack-v5/pull/289/commits/381b840bf043e8c02331fb0d95ff4e19e679093e)
* [Decrease RTP packet length](https://github.com/alienatedsec/yi-hack-v5/pull/289/commits/f5e97ce30d01cf6b41081c7f2a4ca3741f19bab9)
* [Improve sync/async read](https://github.com/alienatedsec/yi-hack-v5/pull/289/commits/9897afc09df6e08870401341e53262bc5a3be558)
* [Update gitignore](https://github.com/alienatedsec/yi-hack-v5/pull/289/commits/d29169e3cceb7765e9ffb182a3663b3ec51bfc0b)
* [New features: disable, force low and already running](https://github.com/alienatedsec/yi-hack-v5/pull/289/commits/70a78bd9ba275af44f93c4d11928b92af4a6d87c)
* [Remove unused settings from config files](https://github.com/alienatedsec/yi-hack-v5/pull/289/commits/d19971d4933f1addc75192ad786e3a120ba02729)
* [Add options to disable snapshot and force low res](https://github.com/alienatedsec/yi-hack-v5/pull/289/commits/c7034e24c88b105183cdb1b802ec9d4445b6d963)
* [Remove unused mqtt options](https://github.com/alienatedsec/yi-hack-v5/pull/289/commits/fcb3bc313482d30c3293fb8a5830f678ba4ba565)
* [Fix and improve h264grabber](https://github.com/alienatedsec/yi-hack-v5/pull/289/commits/2dc4cfa895413867f7bdc3b71e594c322d47401c)
* [Fix system.conf parameters](https://github.com/alienatedsec/yi-hack-v5/pull/289/commits/6ce47e243809f55bf4df60f9f5b5eadec930b2ad)
* [Remove ai_human_detection and restore baby_crying](https://github.com/alienatedsec/yi-hack-v5/pull/289/commits/a309b2aac774f2cb39f34969d7721f9c6df2a02a)
* [Wi-Fi Recovery Partition](https://github.com/alienatedsec/yi-hack-v5/commit/25db09f312d3f05f5ed64c81ec9f4ad5a5167f64) - Instructions in progress
* [Change size of FIFO queues](https://github.com/alienatedsec/yi-hack-v5/pull/299/commits/ee0da9d0eb8afdb560953dca9d7f9ac2631d8f92)
* [Restore audio stream ch0_2.h264](https://github.com/alienatedsec/yi-hack-v5/pull/299/commits/66030f5abbbd86f186907809cb51c0a3a0af898f)
* [Add native API to take low-res snapshot](https://github.com/alienatedsec/yi-hack-v5/pull/299/commits/cf13502b466fd084cfa7d442ca128056ab7540e8)
* [Remove option: watermark for onvif snapshot](https://github.com/alienatedsec/yi-hack-v5/pull/299/commits/15e07ff6e9e2d5d39af4fb8da58cb7c325cf32cb)
* [Change label for "Force low res snapshot"](https://github.com/alienatedsec/yi-hack-v5/pull/299/commits/c821423f61da34d030435b4081c5290e9221b5f4)
* [Improve RTSP server](https://github.com/alienatedsec/yi-hack-v5/pull/299/commits/54b1e653d8aa6136537568b3d8a4b8d7b66bb7cb)
* [Minor fixes](https://github.com/alienatedsec/yi-hack-v5/pull/299/commits/e6ebbce764ecd559ec5d612619caba1f9197b154)
* [Remove unused options when empty password](https://github.com/alienatedsec/yi-hack-v5/pull/299/commits/9c4fff63880b10779c4a4802f22503fb29cd61d6)
* [Add mqttv4 to watchdog](https://github.com/alienatedsec/yi-hack-v5/pull/299/commits/434db6d2ac1c70da9d7895087fcbf48f7fe562a5)
* [Add "if supported" in camera settings page](https://github.com/alienatedsec/yi-hack-v5/pull/299/commits/3cedddd5ab5d40d8ea432a1ae7869b4ed6a3f541)
* [Add support for native time osd](https://github.com/alienatedsec/yi-hack-v5/pull/299/commits/b305a15a5cf5b99cec55f03058e7d8635c731232)
* [[Change] launch.sh optimised to kill gently](https://github.com/alienatedsec/yi-hack-v5/commit/63213ca4b62245b752d05ecb25d957f983c986f0) 
* [[Change] Tuning launch.sh](https://github.com/alienatedsec/yi-hack-v5/commit/26009c1e3c4e2fbe635ea7f601a1b7776c7e85c9) 
* [[New] Compile specific module - thanks](https://github.com/alienatedsec/yi-hack-v5/commit/db856f3dbde9e091cd52cf74411d363c39203347) @thelittlefireman
* `wpa_supplicant` upgrade to `v2.10` - this was an attempt to prevent boot loop issues experienced with certain Wi-Fi configurations (e.g., multiple APs with the same SSID) or any modern Wi-Fi networks - for example, a mixed mode WPA/WPA2 is subject to be removed in the future `wpa_supplicant` releases 
* Initial support for WPA3-Personal - Simultaneous Authentication of Equals (SAE)
* Support to encrypt your Wi-Fi password with `wpa_passphrase`
* Fixing cloudAPI workflow for CN-based cameras
* Removed duplicate configs
* Merging @roleoroleo 's work onto pre-release in https://github.com/alienatedsec/yi-hack-v5/pull/330 and https://github.com/alienatedsec/yi-hack-v5/pull/331 - too many changes to list them all here
* fix: WARNING in HA => MQTT entity name starts with the device name in… by @Minims in https://github.com/alienatedsec/yi-hack-v5/pull/327
* fix: mqtt disconnecting immediately after boot by @fidesachates in https://github.com/alienatedsec/yi-hack-v5/pull/328
* Fix for #334 - reduced the image size by compressing multiple large files in the `home` partition 
* Fix for #336 - added missing config thanks @lebestnoob
* Fix for `yi_home` and `yi_dome` - the `wpa_supplicant` upgrade in the baseline image was prevented by the YI upgrade of an older file
* Recovered lost `system.sh` upgrades and other functions 
* Cleared the libs folder on the `home` partition - attempt to fix #312 and #291 
* The `Busybox` was upgraded to `1.36.1` and its size was reduced to address #312
* Moved some tools from `rootfs` to `/tmp/sd/yi-hack-v5/.../[s]bin` folders using `busybox_tools` submodule - again to address #312 
* Fixed Curl SSL crypto library 
* Fixed symlink to VI #346 
* Fixed the MQTT issue #333 
* Fixed no watermark in streams #336 - thanks @Minims 
* Amended config reading - seems to work quicker - thanks @Minims 
* Added file versioning - so that we can track changes without upgrading the baseline
* Improve MQTT by @Minims in https://github.com/alienatedsec/yi-hack-v5/pull/349 - please test if working as intended
* Fixed WiFi scan page
* Improved the `wpa_supplicant` upgrade logic
* Reduced `busybox_tools` size on the SD

## New Contributors
* @Minims made their first contribution in https://github.com/alienatedsec/yi-hack-v5/pull/327
* @fidesachates made their first contribution in https://github.com/alienatedsec/yi-hack-v5/pull/328
* @roleoroleo made so many contributions - thanks for the additional effort

**Full Changelog**: https://github.com/alienatedsec/yi-hack-v5/compare/0.4.0...0.4.1

<hr /><em>This discussion was created from the release <a href='https://github.com/alienatedsec/yi-hack-v5/releases/tag/0.4.1'>0.4.1</a>.</em>

**Comment by reef-actor** [comment]
This looks great, thanks for all your efforts. To clarify, is it necessary to upgrade by FAT32 sd-card method if I am on pre-release 0.4.1e already?

**Reply by alienatedsec** [reply]
Thanks @reef-actor 
yes, please upgrade as per the release notes - the only reason is that version `0.4.1e` is too far behind `0.4.1j` https://github.com/alienatedsec/yi-hack-v5/compare/0.4.1e...0.4.1j which this final version is based on.

**Comment by g1za** [comment]
Following the port change, what is the new location for recorded video files?
I have an automation that used to send the videos via Telegram and I updated the code to reflect the new port but this is the MQTT message I get, which seems to point to the wrong location, as files is empty
```
Message 15 received on yicam/motion_files at 23:06:
{
"start":"2024-02-26T23:04:04+01:00",
"end":"2024-02-26T23:05:11+01:00",
"files":[ ]
}
QoS: 0 - Retain: false
```

I updated the url in my automation from 
```
action:
    - repeat:
        for_each: '{{ (trigger.payload_json.files) | sort() }}'
        sequence:
          - service: telegram_bot.send_video
            data:
              target: !secret telegram_chat_id
              username: !secret yi_camera_user
              password: !secret yi_camera_pwd
              url: "http://192.168.188.7:8080/record/{{ repeat.item }}"
```

to

```
action:
    - repeat:
        for_each: '{{ (trigger.payload_json.files) | sort() }}'
        sequence:
          - service: telegram_bot.send_video
            data:
              target: !secret telegram_chat_id
              username: !secret yi_camera_user
              password: !secret yi_camera_pwd
              url: "http://192.168.188.7/record/{{ repeat.item }}"
```

which seems correct; but given the MQTT message maybe the files are not being retrieved at all? If needed I can open a bug.
Thanks!

EDIT:
Also this one seems to be empty:
```
Message 17 received on yicam/motion_detection_image at 23:16:

QoS: 1 - Retain: false
 ```

**Reply by alienatedsec** [reply]
Is your SD formatted in the YI app to record videos?

**Reply by g1za** [reply]
Not sure this time. Let me try.

**Reply by g1za** [reply]
I can't connect from the app so I need to wipe everything, install stock firmware and then the hack. But FYI the videos are being recorded, I can see them from the Events tab.

**Reply by g1za** [reply]
I reinstalled the hack, formatted the SD from the app. At that point the yi-hack folder was missing and I was unable to access the web interface. I copied that folder again but the MQTT message on recorded events is still the same (empty) while I can see the recorded video on the events tab. Should I open a ticket?

```Messaggio 42 ricevuto su yicam/motion_files alle 00:32:

{
"start":"2024-02-27T00:30:32+01:00",
"end":"2024-02-27T00:31:46+01:00",
"files":[ ]
}

QoS: 0 - Retain: false ```

**Reply by alienatedsec** [reply]
Apologies @g1za but the new Busybox must have "broken" a certain way of working. Its not a bug as such, and I guess the way you used it before had some security implications.

**Reply by g1za** [reply]
I see but it seems strange to me this had security implications. In the end it was just listing the name of the files. Are we sure it is pointing to the new location (port 80, instead of 8080) to retrive filenames? Would that explain why the list is empty?

**Reply by alienatedsec** [reply]
I will check as soon as I can. @g1za

**Reply by roleoroleo** [reply]
It could be a timezone issue.
I've had a similar problem in the past.
I will check it.

**Comment by bonjour81** [comment]
Hi, I'm a bit confuse on how to upgrade from 0.3.2  ?

**Reply by alienatedsec** [reply]
@bonjour81 please follow the official guideline on the main page.

**Comment by iSeBzh** [comment]
Hello, I formatted the SD card. I left the card's name as “Untitled.” I installed the home_y18 and rootfs_18 files, as well as the yi-hack-v5 directory, at the root of the SD card. I didn't make any changes. When I connect using the IP address on port 80, I see a login screen asking for a password.
